go/vt/vtgate/planbuilder/insert.go

```diff
@@ -209,6 +210,10 @@ func buildInsertSelectPlan(ins *sqlparser.Insert, table *vindexes.Table, reserve
 	}
 	eins.Input = plan
 
+	if strings.Contains(plan.GetTableName(), table.Name.String()) {
+		eins.ForceNonStreaming = true
+	}
```

Comment:
We should add a comment here

Reply:
I added the comment just above this condition.

Reply:
just realised I did not push that.